From: George Kashperko <george@znau.edu.ua>

Make SSB AI get selected on bcm47xx boot over bcma bus driver.
Signed-off-by: George Kashperko <george@znau.edu.ua>
---
 arch/mips/bcm47xx/setup.c |   11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)
--- linux-3.3.8.orig/arch/mips/bcm47xx/setup.c	2012-12-08 01:54:58.736618898 +0200
+++ linux-3.3.8/arch/mips/bcm47xx/setup.c	2012-12-08 01:57:02.990610404 +0200
@@ -235,11 +235,20 @@ static void __init bcm47xx_register_bcma
 }
 #endif
 
+static inline bool use_ssb_ai(void)
+{
+#ifdef CONFIG_SSB_BUS_AI
+	return true;
+#else
+	return false;
+#endif
+}
+
 void __init plat_mem_setup(void)
 {
 	struct cpuinfo_mips *c = &current_cpu_data;
 
-	if (c->cputype == CPU_74K) {
+	if (c->cputype == CPU_74K && !use_ssb_ai()) {
 		printk(KERN_INFO "bcm47xx: using bcma bus\n");
 #ifdef CONFIG_BCM47XX_BCMA
 		bcm47xx_bus_type = BCM47XX_BUS_TYPE_BCMA;
